### PR TITLE
Add an option to authenticate with token

### PR DESCRIPTION
**Pull request by bergzand**
An optional token config option is added to the configuration. When
used, the script will authenticate with the token. The matrixClient
class requires the full ID of the user when using this.

### CODE REVIEW

File: matrix.py [hunk 1]
```diff
@@ -82,10 +82,16 @@ def merge_config(args, config):
 def setup(config):
     """Sets up the Matrix client. Makes sure the (specified) room is joined.
     """
+    loginargs = {}
+    if 'token' in config:
+        loginargs['user_id'] = config['username']
```
Comment (lordievader) [inline matrix.py]: This line gives an error: `ValueError: UserIDs start with @`.
Could you fix this by changing this line to:
```
loginargs['user_id'] = '@{0}:{1}'.format(
    config['username'],
    config['domain'])
```
PEP8 likes a newline after 87 :wink:

File: matrix.py [hunk 2]
```diff
@@ -82,10 +82,16 @@ def merge_config(args, config):
...
+        loginargs['token'] = config['token']
     client = MatrixClient("https://{0}:{1}".format(
-        config['homeserver'], int(config['port'])))
-    client.login_with_password(
-        username=config['username'], password=config['password'])
+        config['homeserver'], int(config['port'])), **loginargs)
+    if 'token' not in config:
+        token = client.login_with_password(
+            username=config['username'], password=config['password'])
+        logging.info("Authenticated, received token: \"%s\"", token)
```
Comment (lordievader) [inline matrix.py:97]: PEP8 likes a newline after 94 :wink:

**Review by lordievader — CHANGES_REQUESTED** [review]
Looks good, if you could make a few minor changes I'll merge them into master.

File: matrix.py (hunk 1 above)
Comment (bergzand) [reply matrix.py]: Fixed

File: matrix.py (hunk 2 above)
Comment (bergzand) [reply matrix.py:97]: Also fixed

**Comment by lordievader** [comment]
Merged!

**Pull request closed without merging**